src/pykeen/nn/representation.py

```diff
@@ -1053,3 +1055,241 @@ def __init__(self, labels: Sequence[str], **kwargs):
+    >>> labels = {1: "a first description", 4: "a second description"}
+    >>> label_initializer = init.LabelBasedInitializer(labels=list(labels.values()))
+    >>> shape = label_initializer.tensor.shape[1:]
+    >>> label_repr = Embedding(max_id=len(labels), shape=shape, initializer=label_initializer, trainable=False)
+
+    Next, we directly create representations for the remaining ones using the backfill representation.
+    To do this, we need to create an iterable (e.g., a set) of all of the entity IDs that are in the base
+    representation. Then, the assignments to the base representation and an auxillary representation are
+    automatically generated for the base class
+
+    >>> from pykeen.nn import BackfillRepresentation
+    >>> entity_repr = BackfillRepresentation(assignment=set(labels), max_id=num_entities, base=label_repr)
+
+    For brevity, we use here randomly generated triples factories instead of the actual data
+    >>> from pykeen.triples.generation import generate_triples_factory
+    >>> training = generate_triples_factory(num_entities=num_entities, num_relations=5, num_triples=31)
+    >>> testing = generate_triples_factory(num_entities=num_entities, num_relations=5, num_triples=17)
+    The combined representation can now be used as any other representation, e.g., to train a DistMult model:
+    >>> from pykeen.pipeline import pipeline
+    >>> from pykeen.models import ERModel
+    >>> pipeline(
+    ...     model=ERModel,
+    ...     interaction="distmult",
+    ...     model_kwargs=dict(
+    ...         entity_representation=entity_repr,
+    ...         relation_representation_kwargs=dict(shape=shape),
+    ...     ),
+    ...     training=training,
+    ...     testing=testing,
+    ... )
+    """
+
+    def __index__(
+        self,
+        max_id: int,
+        assignment: Iterable[int],
+        base: Hint[Representation] = None,
+        base_kwargs: OptionalKwargs = None,
+        embedding_kwargs: OptionalKwargs = None,
+        **kwargs,
+    ):
+        """Initialize the representation.
+
+        :param max_id:
+            The total number of entities that need to be embedded
+        :param assignment:
+            An iterable of integer entity indexes that need to be
+        :param base:
+            the base representation, or hints thereof.
+        :param base_kwargs:
+            keyword-based parameters to instantiate the base representation
+        :param kwargs:
+            additional keyword-based parameters passed to :meth:`Representation.__init__`. May not contain `max_id`,
+            or `shape`, which are inferred from the base representations.
+        """
+        # import here to avoid cyclic import
+        from . import representation_resolver
+
+        assignment = set(assignment)
+        base = representation_resolver.make(base, base_kwargs, max_id=len(assignment))
+        embedding = Embedding(max_id=max_id - base.max_id, shape=base.shape[1:], **(embedding_kwargs or {}))
```

Comment:
fix shape?

Reply:
https://github.com/pykeen/pykeen/pull/980/commits/37a81f3f4e7a97146d85fc62dbbf106e1e0009d6

the shape of a `Representation` module is the individual shape. For the example in PartitionRepresentation, we accessed the shape of the full initial tensor, which has a leading `max_id` dimension.